sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSquidSensor.java

```diff
@@ -79,9 +84,66 @@ public void execute(SensorContext context) {
     sonarComponents.setSensorContext(context);
     sonarComponents.registerCheckClasses(CheckList.REPOSITORY_KEY, CheckList.getJavaChecks());
     sonarComponents.registerTestCheckClasses(CheckList.REPOSITORY_KEY, CheckList.getJavaTestChecks());
-    Measurer measurer = new Measurer(fs, context, noSonarFilter);
-    JavaSquid squid = new JavaSquid(getJavaVersion(), sonarComponents, measurer, javaResourceLocator, postAnalysisIssueFilter, sonarComponents.checkClasses());
-    squid.scan(getSourceFiles(), getTestFiles());
+    try {
+      ExecutorService executorService = createExecutor();
+      CompletionService cs = new ExecutorCompletionService(executorService);
+      ThreadContext threadContext = new ThreadContext(context);
+      int fileSize = 0;
+      for (File sourceFile : getSourceFiles()) {
+        cs.submit(() -> scan(threadContext, sourceFile, false), null);
+        fileSize++;
+      }
+      for (File testFile : getTestFiles()) {
+        cs.submit(() -> scan(threadContext, testFile, true), null);
+        fileSize++;
+      }
+      executorService.shutdown();
+      for (int i = 0; i < fileSize; i++) {
+        try {
+          cs.take().get();
+          if (i % 10 == 0) {
+            LOG.info("analyzed " + i + "/" + fileSize + " files");
+          }
+        } catch (InterruptedException e) {
+          e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148)